File: services/gameService/GameRoom.java
package services.gameService;

import java.util.ArrayList;
import javax.swing.JPanel;

/**
 *
 * @author Jean Carlos Santoya Cabrera jeancarlosodi@gmail.com
 */
public abstract class GameRoom extends JPanel{
    
    private ArrayList<GameObject> objects;
    private String title;
    
    public GameRoom (int x, int y, int width, int height, String title){
        objects = new ArrayList<GameObject>();
        
        this.title = title;
        
        setIgnoreRepaint(true);
        setLocation(x*-1, y*-1);
        setSize(width, height);
        setLayout(null);
    }
    
    public void addGameObject(GameObject object, int height){
        objects.add(object);
        add(objects.get(objects.size() - 1), height);
    }
    
    public void addGameObject(GameObject object){
        objects.add(object);
        add(objects.get(objects.size() - 1));
    }
    
    public void addTiles(int x, int y, int filas, int columnas, GameTile tile){
        for(int i=0; i<filas; i++){
            for(int j=0; j<columnas; j++){
                //Attempts
                //add(new Class<tile.getClass()>(x+j*64,y+i*64));
                //add(tile.getClass().newInstance().setLocation(x+j*64,y+i*64));
            }
        }
    }
    
    public void run(){//this method runs 60 times per second
        setVisible(false);
        if(objects.size() > 0){
            for(int i=0; i<objects.size(); i++){
                objects.get(i).run();//moves my JLabels with setLocation
            }
        }
        setVisible(true);
        repaint();
    }
    
    public String getTitle(){
        return title;
    }
}

File: services/gameService/GameRoom.kt
package services.gameService

import java.util.*
import javax.swing.JPanel

/**
 *
 * @author Jean Carlos Santoya Cabrera jeancarlosodi@gmail.com
 */
abstract class GameRoom(x: Int, y: Int, width: Int, height: Int, title: String) : JPanel() {
    private val objects: ArrayList<GameObject>
    val title: String
    fun addGameObject(`object`: GameObject, height: Int) {
        objects.add(`object`)
        add(objects[objects.size - 1], height)
    }

    fun addGameObject(`object`: GameObject) {
        objects.add(`object`)
        add(objects[objects.size - 1])
    }

    fun addTiles(x: Int, y: Int, filas: Int, columnas: Int, tile: services.gameService.GameTile?) {
        for (i in 0 until filas) {
            for (j in 0 until columnas) {
                //Attempts
                //add(new Class<tile.getClass()>(x+j*64,y+i*64));
                //add(tile.getClass().newInstance().setLocation(x+j*64,y+i*64));
            }
        }
    }

    fun run() { //this method runs 60 times per second
        isVisible = false
        if (objects.size > 0) {
            for (i in objects.indices) {
                objects[i].run() //moves my JLabels with setLocation
            }
        }
        isVisible = true
        repaint()
    }

    init {
        objects = ArrayList()
        this.title = title
        ignoreRepaint = true
        setLocation(x * -1, y * -1)
        setSize(width, height)
        layout = null
    }
}